### PR TITLE
gcc7 compilation warnings fix in RecoJets pkgs

### PR DESCRIPTION
**Pull request by mrodozov**
Fixes for compilation warnings when compiling with gcc7 and more flags listed here:
https://cmssdt.cern.ch/SDT/cgi-bin/showBuildLogs.py/slc6_amd64_gcc700/www/mon/9.3-mon-23/CMSSW_9_3_X_2017-07-24-2300
packages 18 and 19 comp warnings

### CODE REVIEW

File: RecoJets/FFTJetProducers/plugins/FFTJetEFlowSmoother.cc [hunk 1]
```diff
@@ -34,8 +34,8 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 
 #include "DataFormats/Common/interface/View.h"
-#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
-
+//#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
```
Comment (perrotta) [inline RecoJets/FFTJetProducers/plugins/FFTJetEFlowSmoother.cc]: Please remove commented out code, if there is no reason to keep it

File: RecoJets/JetAnalyzers/src/FFTJetImageRecorder.cc [hunk 2]
```diff
@@ -33,8 +33,9 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 
 #include "DataFormats/Common/interface/Handle.h"
-#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
-
+//#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
```
Comment (perrotta) [inline RecoJets/JetAnalyzers/src/FFTJetImageRecorder.cc]: Please remove commented out code, if there is no reason to keep it

File: RecoJets/JetAnalyzers/src/FFTJetPileupAnalyzer.cc [hunk 3]
```diff
@@ -31,7 +31,9 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 
 #include "DataFormats/Common/interface/Handle.h"
-#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
+//#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
```
Comment (perrotta) [inline RecoJets/JetAnalyzers/src/FFTJetPileupAnalyzer.cc]: Please remove commented out code, if there is no reason to keep it

**Comment by perrotta** [comment]
please test

File: RecoJets/JetAnalyzers/src/FFTJetImageRecorder.cc [hunk 4]
```diff
@@ -33,8 +33,8 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 
 #include "DataFormats/Common/interface/Handle.h"
-#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
-
+#include <TH3F.h>
+#include <TH2D.h>
```
Comment (slava77) [inline RecoJets/JetAnalyzers/src/FFTJetImageRecorder.cc]: is TH2D really necessary?
I only see TH3F mentioned in the file

Comment (mrodozov) [inline RecoJets/JetAnalyzers/src/FFTJetImageRecorder.cc]: It wasn't. The files are updated accordingly

File: RecoJets/JetAnalyzers/src/FFTJetPileupAnalyzer.cc [hunk 5]
```diff
@@ -31,7 +31,8 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 
 #include "DataFormats/Common/interface/Handle.h"
-#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
+#include <TH3F.h>
```
Comment (perrotta) [inline RecoJets/JetAnalyzers/src/FFTJetPileupAnalyzer.cc]: Is TH3F really needed? I only see TH2D in the code

File: DQM/L1TMonitor/src/L1TRate.cc [hunk 6]
```diff
@@ -24,7 +24,7 @@
 #include "CondFormats/DataRecord/interface/L1GtTriggerMenuRcd.h"
 #include "CondFormats/DataRecord/interface/L1GtPrescaleFactorsAlgoTrigRcd.h"
 
-#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
+//#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
```
Comment (perrotta) [inline DQM/L1TMonitor/src/L1TRate.cc]: Please remove commented out code, or add another line of comment which explains why it must be kept.

File: DQM/L1TMonitorClient/src/L1TOccupancyClient.cc [hunk 7]
```diff
@@ -8,7 +8,7 @@
 #include "DQMServices/Core/interface/QReport.h"
 #include "DQMServices/Core/interface/DQMStore.h"
 #include "DQMServices/Core/interface/MonitorElement.h"
-#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
+//#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
```
Comment (perrotta) [inline DQM/L1TMonitorClient/src/L1TOccupancyClient.cc]: Please remove commented out code, or add another line of comment which explains why it must be kept.

File: DQM/L1TMonitorClient/src/L1TTestsSummary.cc [hunk 8]
```diff
@@ -9,7 +9,7 @@
 #include "DQMServices/Core/interface/DQMStore.h"
 #include "DQMServices/Core/interface/MonitorElement.h"
 #include "DQMServices/Core/interface/DQMChannel.h"
-#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
+//#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
```
Comment (perrotta) [inline DQM/L1TMonitorClient/src/L1TTestsSummary.cc]: Please remove commented out code, or add another line of comment which explains why it must be kept.

File: DQMOffline/L1Trigger/src/L1TMuonDQMOffline.cc [hunk 9]
```diff
@@ -11,7 +11,7 @@
 #include "DataFormats/L1TMuon/interface/RegionalMuonCandFwd.h"
 #include "DataFormats/L1Trigger/interface/Muon.h"
 #include "DQMServices/Core/interface/DQMStore.h"
-#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
+//#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
```
Comment (perrotta) [inline DQMOffline/L1Trigger/src/L1TMuonDQMOffline.cc]: Please remove commented out code, or add another line of comment which explains why it must be kept.

File: DQMOffline/L1Trigger/src/L1TRate_Offline.cc [hunk 10]
```diff
@@ -10,7 +10,7 @@
 #include "CondFormats/DataRecord/interface/L1GtTriggerMenuRcd.h"
 #include "CondFormats/DataRecord/interface/L1GtPrescaleFactorsAlgoTrigRcd.h"
 
-#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
+//#include "DataFormats/Histograms/interface/MEtoEDMFormat.h"
```
Comment (perrotta) [inline DQMOffline/L1Trigger/src/L1TRate_Offline.cc]: Please remove commented out code, or add another line of comment which explains why it must be kept.

**Comment by davidlange6** [comment]
please test

**Comment by mrodozov** [comment]
abort test

**Comment by mrodozov** [comment]
please test

**Comment by smuzaffar** [comment]
please test
last IB is broken which is not ignored for PR tests

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/19924	6017761
- technical update to avoid including DataFormats/Histograms/interface/MEtoEDMFormat.h which is not needed in full
- jenkins tests pass

**Comment by davidlange6** [comment]
+1

**Pull request merged**